crates/egui-wgpu/Cargo.toml

```diff
@@ -31,9 +31,15 @@ all-features = true
 ## Enable profiling with the [`puffin`](https://docs.rs/puffin) crate.
 puffin = ["dep:puffin"]
 
-## Enable [`winit`](https://docs.rs/winit) integration.
+## Enable [`winit`](https://docs.rs/winit) integration. On Linux, requires either `wayland` or `x11`
 winit = ["dep:winit"]
 
+## Enables Wayland support for winit.
+wayland = ["winit?/wayland"]
+
+## Enables x11 support for winit.
+x11 = ["winit?/x11"]
+
```

Comment:
If a user is using `egui-wgpu` with the `winit` feature, they are also depending directly on `winit`, and can set the `x11/wayland` features themselves, no?

Reply:
Yes, but then you have to think about that when running `cargo check`, `cargo doc` etc, which is the problem noted in the issues. Technically you don't have to enable these features in a real project, they just help make the repo more ergonomic to work with. I don't know if there is a better solution.

Reply:
ah I see

Reply:
The problem would still happen with `cargo check` though, just not with `cargo check --all-features` ?